### ISSUE TITLE
When root indexName is absent, `scopedResults` are empty

### ISSUE THREAD

**Issue by marcelgerber** [algolia/instantsearch#6132]
### 🐛 Current behavior

If I create a `<InstantSearch indexName={undefined}>` element (as enabled by #5590 a year ago), with several `<Index>` components inside it, and then want to access `useInstantSearch()`'s `scopedResults` element, I find that it will always be (essentially) empty.

### 🔍 Steps to reproduce

See the repro link below; click the checkbox at the very top and see how the available `indexId`s change.

### Live reproduction

https://codesandbox.io/p/sandbox/algolia-no-root-index-scopedresults-w2jfvl?file=%2Fsrc%2FApp.tsx

### 💭 Expected behavior

I should find entries in `scopedResults` for all sub-indexes, as is the case if a root index name is given.

### Package version

Algolia for JavaScript (4.23.3); Browser (lite); instantsearch.js (4.66.1); react (18.1.0); react-instantsearch (7.7.1); react-instantsearch-core (7.7.1); JS Helper (3.17.0)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by marcelgerber** [comment]
So, I could trace this down to these lines just now:
https://github.com/algolia/instantsearch/blob/1b683c5ebbcae4a77a43ff2034da406c9eb0dbc4/packages/react-instantsearch-core/src/lib/useSearchResults.ts#L26-L35

In the above case, `results` is always `null` (makes sense, I guess), but `searchIndex.getScopedResults()` is not. But because of this if-statement, it is never updated.

**Comment by Haroenv** [comment]
That's a good point, although I'm not sure how to change this condition without then also making people deal with `null` for the initial render (which people usually don't have to care about, and otherwise destructures badly). An option could be a way to go, but isn't that nice. Do you have a solution in mind?

**Comment by marcelgerber** [comment]
Hm, I don't know about the internals of InstantSearch and when results are arriving, but can we detect that there is no main index present?
In that case, `results` will always be `null` (and has always been), and it makes sense to update `scopedResults` directly.

**Comment by Haroenv** [comment]
Yes, that's a good point, so in that case the condition could be:

```js
if (results !== null) {
      setSearchResults({ 
       results, 
       scopedResults: searchIndex.getScopedResults(), 
     }); 
} else if (search.mainIndex.getIndexName() === undefined) {
      setSearchResults({ 
       results: getIndexSearchResults(searchIndex), 
       scopedResults: searchIndex.getScopedResults(), 
     }); 
}
```

If you're interested in seeing this through, we'd be happy to accept a pull request with this feature and new tests :)

**Comment by dhayab** [comment]
Hi, we just released react-instantsearch@7.7.3 and react-instantsearch-core@7.7.3 which include a fix for this issue.

**Comment by marcelgerber** [comment]
Thank you @dhayab, works perfectly and flawlessly!
